Resolve Keras File Format Issues

TensorFlow updated to version 2.16.1 on March 8, 2024. Python 3.9+ tests in the main idaes-pse repository that install the latest package and run examples now throw errors similar to the message below:

```
E           ValueError: File format not supported: filepath=D:\a\idaes-pse\idaes-pse\idaes\core\surrogate\plotting\tests\keras_surrogate. Keras 3 only supports V3 `.keras` files and legacy H5 format files (`.h5` extension). Note that the legacy SavedModel format is not supported by `load_model()` in Keras 3. In order to reload a TensorFlow SavedModel as an inference-only layer in Keras 3, use `keras.layers.TFSMLayer(D:\a\idaes-pse\idaes-pse\idaes\core\surrogate\plotting\tests\keras_surrogate, call_endpoint='serving_default')` (note that your `call_endpoint` might have a different name).
```

Examples that use Keras should be updated accordingly.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -57,7 +57,7 @@ gui = [
 omlt = [
     # For Keras/OMLT
     "omlt",
-    "tensorflow",
+    'tensorflow<2.16.1 ; python_version < "3.12"',
 ]
 idaes = [
     "idaes-pse",  # installing IDAES (from release) is opt-in

